automation/test.sh

```diff
@@ -204,14 +205,7 @@ set -e
 echo "Nodes are ready:"
 kubectl get nodes
 
-make cluster-build
-
-# I do not have good indication that OKD API server ready to serve requests, so I will just
-# repeat cluster-deploy until it succeeds
-until make cluster-deploy; do
-    sleep 1
-done
-
```

Comment:
@oshoval if you will adopt the suggested re-design this can be kept since the parallel logic will occur at cluster-build target

Reply:
i dont think we should keep it anyhow, the change i introduce is cleaner, adds make cluster-sync to the pipeline of testing ,and not its build stones.
this code was introduce just for okd providers as the comment say, and we dont have them anymore

Reply:
@oshoval Regarding OKD, there was a discussion on the last community meeting to make it alive again. But anyway, you are right, deleting this indeed introduce a more clean code, the only nit is that this cleanup looks out of the PR context.

Reply:
the cleanup was needed in order to use the new feature as part of the CI,
this way it will also be tested and not left forgotten